### PR TITLE
#12025 - Add an environment list for the mobile app

### LINKED ISSUE #12025: Add an environment list for the mobile app [3]
### Feature Description

After an Environment entity has been created in https://github.com/SORMAS-Foundation/SORMAS-Project/issues/11796 as part of epic https://github.com/SORMAS-Foundation/SORMAS-Project/issues/11533, a list that contains those entities is now required for the mobile app.

### Added Value/Benefit

Users will be able to see and filter a list of the environment entities created in the system.

### Acceptance Criteria

- [x] A new menu entry called "Environments" has been added with an appropriate icon (see the mockup below for inspiration); the entry has been placed directly below "Samples"
- [x] The entry is only visible when the feature added in #11572 is enabled and the user has the `ENVIRONMENT_VIEW` user right
- [x] The items in the list includes the following information:
  - ID (= the UUID of the environment entity)
  - External ID
  - Name
  - Environment media
  - GPS lat
  - GPS lon
- [x] The list is automatically sorted by change date
- [x] The user can use the hovering button in the bottom right to filter the list by investigation status

### Implementation Details

_No response_

### Mockups

_No response_

### Additional Information

_No response_

### PR DESCRIPTION
**Pull request by sergiupacurariu**
Fixes #12025

### CODE REVIEW

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12344

**Review by leventegal-she — CHANGES_REQUESTED** [review]
I get `java.sql.SQLException: Unknown field 'wateruse' from the Android sqlite cursor, not in:[id, uuid, changeDate, creationDate, lastOpenedDate, localChangeDate, modified, snapshot, reportDate, reportingUser_id, environmentName, description, externalId, responsibleUser_id, investigationStatus, environmentMedia, waterType, otherWaterType, infrastructureDetails, otherInfrastructureDetails, waterUse, otherWaterUse, location_id]` when I try to open the environment list in the app

File: sormas-app/app/src/main/java/de/symeda/sormas/app/rest/SynchronizeDataAsync.java [hunk 1]
```diff
@@ -144,7 +147,7 @@ protected Void doInBackground(Void... params) {
 
 		try {
 			boolean pullAndRemoveObsoleteNecessary = ConfigProvider.getLastObsoleteUuidsSyncDate() == null
-				|| DateHelper.getFullDaysBetween(ConfigProvider.getLastObsoleteUuidsSyncDate(), new Date()) >= 1;
+				|| DateHelper.getFullDaysBetween(ConfigProvider.getLastObsoleteUuidsSyncDate(), new Date()) >= 0;
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/rest/SynchronizeDataAsync.java:150]: changed back to 1

File: sormas-app/app/src/main/java/de/symeda/sormas/app/util/DateFormatHelper.java [hunk 2]
```diff
@@ -47,4 +48,12 @@ public static String formatDateInterval(Date startDate, Date endDate) {
 		}
 		return intervalDate;
 	}
+
+	public static String getLatLonLocation(Location location){
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/util/DateFormatHelper.java:52]: method moved to ViewHelper class

File: sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java [hunk 3]
```diff
@@ -3050,10 +3054,10 @@ public void onUpgrade(SQLiteDatabase db, ConnectionSource connectionSource, int
 
 			case 344:
 				currentVersion = 344;
-				getDao(DiseaseConfiguration.class).executeRaw(
-					"CREATE TABLE environments(id integer primary key autoincrement, uuid VARCHAR(36) NOT NULL, "
+				getDao(Environment.class).executeRaw(
+					"CREATE TABLE environments(id integer primary key autoincrement, uuid VARCHAR(36) NOT NULL UNIQUE, "
```
[the remark this answers is not in the document]
Comment (sergiupacurariu) [reply sormas-app/app/src/main/java/de/symeda/sormas/app/backend/common/DatabaseHelper.java:3058]: added new migration for the added changes

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12344

**Review by leventegal-she — APPROVED** [review]

**Comment by sormas-vitagroup** [comment]
SonarCloud analysis: https://sonarcloud.io/dashboard?id=SORMAS-Project&pullRequest=12344

**Pull request merged**